### PR TITLE
dev/financial#23 Don't lookup payment processor for 0 amount as there isn't one defined!

### PR DESCRIPTION
**Pull request by mattwire**
Overview
----------------------------------------
See: https://lab.civicrm.org/dev/financial/issues/23

When submitting a contribution page with 0 amount the payment processor is not set. However, the submission still tries to lookup a payment processor by ID and hits the following e-notice in CRM/Contribute/BAO/Contribution.php: 
$CRM16923AnUnreliableMethodHasBeenUserToDeterminePaymentProcessorFromContributionPage.

Before
----------------------------------------
e-notice in CRM/Contribute/BAO/Contribution.php: 
$CRM16923AnUnreliableMethodHasBeenUserToDeterminePaymentProcessorFromContributionPage when amount=0.

After
----------------------------------------
No e-notice in CRM/Contribute/BAO/Contribution.php: 
$CRM16923AnUnreliableMethodHasBeenUserToDeterminePaymentProcessorFromContributionPage when amount=0.

Technical Details
----------------------------------------
If we have a 0 amount the payment processor ID is not set but we try and look it up anyway.

### CODE REVIEW

**Comment by mattwire** [comment]
test this please

File: api/v3/Contribution.php [hunk 1]
```diff
@@ -533,6 +533,7 @@ function civicrm_api3_contribution_completetransaction(&$params) {
   if (isset($params['payment_processor_id'])) {
     $input['payment_processor_id'] = $params['payment_processor_id'];
   }
+  $input['total_amount'] = CRM_Utils_Array::value('total_amount', $params);
```
Comment (pradpnayak) [inline api/v3/Contribution.php]: What would be the use case to update the total amount for a  contribution?

Comment (mattwire) [reply api/v3/Contribution.php]: @pradpnayak So we're not updating the total_amount but we need to pass it as a parameter so the later functions can check if the amount is 0.

**Comment by mattwire** [comment]
test this please

**Comment by mattwire** [comment]
test this please

File: CRM/Contribute/BAO/Contribution.php [hunk 2]
```diff
@@ -2405,7 +2405,7 @@ public function loadRelatedObjects(&$input, &$ids, $loadAll = FALSE) {
       $ids
     ));
 
-    if (!isset($input['payment_processor_id']) && !$paymentProcessorID && $this->contribution_page_id) {
+    if (!isset($input['payment_processor_id']) && !$paymentProcessorID && $this->contribution_page_id && ($input['total_amount'] > 0)) {
```
Comment (seamuslee001) [inline CRM/Contribute/BAO/Contribution.php]: @mattwire you may need to do a `!empty($imput['total_amount'])` here

**Comment by seamuslee001** [comment]
Test failures look related

**Comment by mattwire** [comment]
@seamuslee001 Looks like all tests are passing

**Comment by mattwire** [comment]
test this please

**Pull request closed without merging**